src/main/java/com/oath/halodb/HaloDBInternal.java

```diff
@@ -73,7 +79,7 @@
 
     private HaloDBInternal() {}
 
-    static HaloDBInternal open(File directory, HaloDBOptions options) throws HaloDBException, IOException {
+    static synchronized HaloDBInternal open(File directory, HaloDBOptions options) throws HaloDBException, IOException {
```

Comment:
Just noticed that this is a static method. ```synchronized``` will not do its work as what we expected